src/sdk/pynni/nni/analysis_utils/sensitivity/torch/sensitivity_analysis.py

```diff
@@ -0,0 +1,257 @@
+                different models may need different dataset/criterion
+                , therefore the user need to cover this part by themselves.
+                val_func take the model as the first input parameter, and
+                return the accuracy as output.
+            sparsities:
+                The sparsity list provided by users.
+            prune_type:
+                The pruner type used to prune the conv layers, default is 'l1',
+                and 'l2', 'fine-grained' is also supported.
+            early_stop:
+                If this flag is set, the sensitivity analysis
+                for a conv layer will early stop when the accuracy
+                drop already reach the value of early_stop (0.05 for example).
+                The default value is 1.0, which means the analysis won't stop
+                until all given sparsities are tested.
+
+        """
+        self.model = model
+        self.val_func = val_func
+        self.target_layer = OrderedDict()
+        self.ori_state_dict = copy.deepcopy(self.model.state_dict())
+        self.target_layer = {}
+        self.sensitivities = {}
+        if sparsities is not None:
+            self.sparsities = sorted(sparsities)
+        else:
+            self.sparsities = np.arange(0.1, 1.0, 0.1)
+        self.sparsities = [np.round(x, 2) for x in self.sparsities]
+        self.Pruner = L1FilterPruner
+        if prune_type == 'l2':
+            self.Pruner = L2FilterPruner
+        elif prune_type == 'fine-grained':
+            self.Pruner = LevelPruner
+        self.early_stop = early_stop
+        self.ori_acc = None  # original accuracy for the model
+        # already_pruned is for the iterative sensitivity analysis
+        # For example, sensitivity_pruner iteratively prune the target
+        # model according to the sensitivity. After each round of
+        # pruning, the sensitivity_pruner will test the new sensitivity
+        # for each layer
+        self.already_pruned = {}
+        self.model_parse()
+
+    @property
+    def layers_count(self):
+        return len(self.target_layer)
+
+    def model_parse(self):
+        for name, submodel in self.model.named_modules():
+            for op_type in SUPPORTED_OP_TYPE:
+                if isinstance(submodel, op_type):
+                    self.target_layer[name] = submodel
+                    self.already_pruned[name] = 0
+
+    def analysis(self, val_args=None, val_kwargs=None, start=0, end=None):
+        """
+        This function analyze the sensitivity to pruning for
+        each conv layer in the target model.
+        If %start and %end are not set, we analyze all the conv
+        layers by default. Users can specify several layers to
+        analyze or parallelize the analysis process easily through
+        the %start and %end parameter.
+
+        Parameters
+        ----------
+            start:
+                Layer index of the sensitivity analysis start
```

Comment:
what is the meaning of "Layer index"?

Reply:
this index is only used internally, right? not exposed to users? if it is also exposed to users, we should explain it in more detail.

Reply:
Start and end is the range of the conv layers to be analyzed. For example, if start and end are set to 1 and 10 respectively, then we only analyze the sensitivity of the first ten conv layers of the model. These parameters are for users to analyze the layers in parallel.

Reply:
how does user know which conv is index 2? my concern is this user interface is not very friendly.

Reply:
how about we use the conv names to identify the start point and end point?

Reply:
we can discuss this part in the meeting.